lib/SparqlEndpointFetcher.ts

```diff
@@ -62,6 +62,45 @@ export class SparqlEndpointFetcher {
       ? (parsedQuery.queryType === 'DESCRIBE' ? 'CONSTRUCT' : parsedQuery.queryType) : "UNKNOWN";
   }
 
+  /**
+   * Get the query type of the given update query.
+   *
+   * This will parse the update query and thrown an exception on syntax errors.
+   *
+   * @param {string} query An update query.
+   * @return {'UNKNOWN' | {
+   *  insertDelete: ('insert' | 'delete' | 'deletewhere' | 'insertdelete')[];
+   *  management: ('copy' | 'move' | 'add' | 'load' | 'create' | 'clear' | 'drop')[]
+   * }} The included update operations.
+   */
+  public getUpdateTypes(query: string): 'UNKNOWN' | {
+    insertDelete: ('insert' | 'delete' | 'deletewhere' | 'insertdelete')[];
+    management: ('copy' | 'move' | 'add' | 'load' | 'create' | 'clear' | 'drop')[]
+  } {
```

Comment:
Instead of an array, could we use a hash? `{ insert: true, delete: true, ... }`
This would slightly improve performance for downstreams libs, as they can avoid `indexOf` checks in favor of indexed access.